media.go

```diff
@@ -190,6 +198,8 @@ func newMediaSource[T, U any](d driver.Driver, r MediaReader[T], p U) MediaSourc
 }
 
 func (pc *producerConsumer[T, U]) start() {
+	startLocalCtx, span := trace.StartSpan(pc.rootCancelCtx, "gostream::producerConsumer::start")
```

Comment:
So here we're creating a span that may not call `span.End()`. This function will only spawn a new goroutine iff `pc.listeners == 1`. You can test this by opening up multiple client streams and streaming at the same time and printing out the value of `pc.listeners`. It should sometimes be `2`. Only one of those will call `defer span.End()`. The other will create the span with this line and return from line 209. You could probably fix this just by calling `span.End()` before the return on line 209. Normally you'd defer but that doesn't look as practical here. The few lines between here and the return will most likely never panic.

Reply:
That's a good catch!